tasks/2017/visual-art.md

```diff
@@ -0,0 +1,220 @@
+  title "visual art: Draw a/an <theme> avatar set.". Include a link to the
+  pull request when you submit your task on the GCI website.
+
+**Making inclusive avatars**
+
+Some tips on making your avatars inclusive and diverse:
+- avoiding humanoid avatars is usually a good idea
+- users often like cute animals or creatures
+- well designed initials are always welcome - if you're creating text-only
+  avatars, create a set for all 24 letters
+- smiley faces and custom emoji are used quite often
+- aliens and monsters allow for a lot of creativity
+- using Zulip brand colors and logo in creative ways can be a base for avatars
+- everyday objects, especially computer-related, might be interesting
+
+If you want to read more on inclusive avatars, here's a [great article](https://uxdesign.cc/design-avatars-that-make-sense-and-be-more-inclusive-in-the-process-d4dd6a486ea6).
+
+*Completion criteria:*
+* Student submitted 4-5 distinct avatars (24 if avatars are text-only) either
+  in SVG format or as 2 PNGs (50x50px and 300x300px) per avatar.
+* The avatars share a theme, e.g. it's a set of animals, monsters or smiley
+faces.
+* The avatars' distinct features are clear in the 50x50px image.
+* The avatars are student's own work, not images downloaded from the Internet.
+
+#### Task Type B: Generate identicons for user avatars
+
+Currently, Zulip's non-custom avatars are  [identicons](https://en.wikipedia.org/wiki/Identicon) automatically generated
+by Gravatar. In this task, you'll generate your own identicons for user avatars.
+
+* Create a directory `generate-avatars/<username>`, where `<username>` is
+  your GitHub username.
+
+* Write a Python 3 script for generating identicons from provided data - allow  
+  it to take as many arguments as provided when running the script, with a
+  minimum of 1. The script should hash the provided data, translate the hash to
+  a 50x50px image and save the file as PNG. Call this file
+  `generate-identicons.py` and add it to the directory.
+
+* Test run your script with different inputs and save 5 different identicons as
+  PNG files (name them `avatar-1.png`, `avatar-2.png` etc.). Add them to the
+  directory.
+
+* Add a `README.md` file to your directory. Describe your approach to hashing
+  the data and translating it to a 50x50px image. Add information on the
+  examples you provided - provide the input data for each of your saved images.
+
+* Add a commit with these files, with commit message "visual art: Add    
+  a script for user identicons.".
+
+* Submit a pull request to the `zulip/zulip-gci-submissions` repository, with
+  title "visual art: Generate identicons for user avatars.". Include a link to
+  the pull request when you submit your task on the GCI website.
+
+**Generating identicons**
+
+Identicons are usually generated by hashing chosen user data (like name or
+email) and using that hash to flip image pixels on and off. For creating an
+image with Python you need to:
```

Comment:
I would add a paragraph giving examples (e.g. default gravatars, with a link) and mention that Zulip is hoping to replace our Gravatar identicons with something that looks nicer and more modern.

Reply:
Eh, maybe this is unnecessary; I'm going to merge as-is and we can add tweaks like this later.